### PR TITLE
#162191234 Register a user

### PR DESCRIPTION
**Pull request by RonKbS**
### What does this PR do?
It enables users to signup to the mydiary application.

### Description of tasks to be completed?
Currently, there is no signup feature for this application.
This PR adds that feature by interacting with the backend application hosted on Heroku.

### How should this be manually tested?
- Run `npm install` to install all dependencies
- Run `npm start`
- Enter the UR `http://localhost:3000/` in your browser.
-Click on the `Create Diary` button, enter user details and click on `create`

### What are the relevant pivotal tracker stories?
[#162191234](https://www.pivotaltracker.com/story/show/162191234)

### Any background context you want to add?
Currently, when a user registers, they receive a toast notification to show they have successfully registered. This will be changed when working on the login feature to redirect someone to the dashboard on signing up.

### CODE REVIEW

File: src/tests/reducers/userReducer.test.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      registerUserError: {},
+      loginError: {},
+    };
+  });
+
+  it('should run initial state', () => {
+    expect(userReducer(initialState, {})).toEqual(initialState);
+  });
+
+  it('should begin the loader when SEND_RESET_LINK_INITIATED is changed to true', () => {
```
Comment (kakaemma) [inline src/tests/reducers/userReducer.test.js]: Kindly change the description to reflect the test

**Review by kakaemma — APPROVED** [review]
LGTM

File: src/reducers/rootReducer.js [hunk 2]
```diff
@@ -1,8 +1,8 @@
 import { combineReducers } from 'redux';
+import userReducer from './userReducer';
 
 const rootReducer = combineReducers({
-// dummy object for reducer
-  user: () => ({}),
+  user: userReducer,
```
Comment (Baalmart) [inline src/reducers/rootReducer.js:5]: Could you look into using the property value shorthand? More like:

import user from './userReducer';

const rootReducer = combineReducers(
{
user,
}
);

More details about this ES6 style guide: https://eslint.org/docs/rules/object-shorthand.html

File: src/tests/actions/userActions.test.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    register()(store.dispatch);
+    await flushAllPromises();
+    expect(store.getActions()).toEqual(
+      [
+        { type: REGISTER_USER_INITIATED, payload: true },
+        { type: REGISTER_USER_SUCCESS, payload: true },
+      ],
+    );
+  });
+  it('should not register a user with wrong details', async () => {
```
Comment (Baalmart) [inline src/tests/actions/userActions.test.js:40]: I like the approach of writing "true" and "false" tests for each test case. Very good testing practice! :)

File: src/tests/components/RegisterUser.test.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+  it('should reload on registering successfully', () => {
+    const nextProps = {
+      registerUserSuccess: true,
+    };
+    wrapper.setProps({ ...nextProps });
+    expect(window.location.reload).toBeCalled();
+  });
+
+  it('should call handleChange on form fill', () => {
```
Comment (Baalmart) [inline src/tests/components/RegisterUser.test.js:60]: Could you also write a test for "should not call handleChange on form fill" If we are to follow the spirit of "true" and "false" tests.

**Pull request merged**